test/e2e/cluster/create.go

```diff
@@ -15,12 +15,14 @@ import (
 )
 
 type TestResources struct {
-	ClusterName       string        `yaml:"clusterName"`
-	ClusterRegion     string        `yaml:"clusterRegion"`
-	ClusterNetwork    NetworkConfig `yaml:"clusterNetwork"`
-	HybridNetwork     NetworkConfig `yaml:"hybridNetwork"`
-	KubernetesVersion string        `yaml:"kubernetesVersion"`
-	Cni               string        `yaml:"cni"`
+	ClusterName       string           `yaml:"clusterName"`
+	ClusterRegion     string           `yaml:"clusterRegion"`
+	ClusterNetwork    NetworkConfig    `yaml:"clusterNetwork"`
+	HybridNetwork     NetworkConfig    `yaml:"hybridNetwork"`
+	KubernetesVersion string           `yaml:"kubernetesVersion"`
+	Cni               string           `yaml:"cni"`
```

Comment:
i dont love what i did here, looking for feedback/other options?  One I thought migth be better than this would be to turn the cni field into the config instead of a string?

As background, the slightly complicating bit here is for calico we need to give it two registry/paths. For ecr when you setup the ptc it has to be namespaced, so `<accountID>.<ecr>.com/docker.io` or `<accountID>.<ecr>.com/quay.io`. In the case of calico, the operator is in quay and the rest are from docker.

Reply:
Here is an example config

```
clusterName: nodeadm-e2e-tests-fc7ea9db-733d-445b-9af5-82fcf1497092
clusterRegion: us-west-2
kubernetesVersion: 1.31
cni: cilium
calico:
  containerRegistry: <acct>.dkr.ecr.us-west-2.amazonaws.com/docker.io
  operatorContainerRegistry: <acct>.dkr.ecr.us-west-2.amazonaws.com/quay.io
cilium:
  containerRegistry: <acct>.dkr.ecr.us-west-2.amazonaws.com/quay.io
clusterNetwork:
  vpcCidr: 10.0.0.0/16
  publicSubnetCidr: 10.0.10.0/24
  privateSubnetCidr: 10.0.20.0/24
hybridNetwork:
  vpcCidr: 10.1.0.0/16
  publicSubnetCidr: 10.1.1.0/24
  privateSubnetCidr: 10.1.2.0/24
  podCidr: 10.2.0.0/16
```

Reply:
Thinking more, if we are willing to break from the current ecr we use today and standardize on `quay.io` `docker.io`, potentially even `ecr-public`, as the namespaces we could just add `containerRegistry` at the root of the spec, not specific to the cni, and just hardcode the namespace/prefixes accordingly.   We could even handle the nginx container this way as well.